convert/lib/convert.py

```diff
@@ -0,0 +1,545 @@
+                        append_rule_sg = rule_sg_r[t(rule_sg)]
+                    resource_id = f"{sg}{t(cf_sgr['rule'])}{append_rule_sg}"
+                elif cf_igw_attachment := item.get("cf_igw_attachment"):
+                    igw_id = resources[t(cf_igw_attachment["igw"])]
+                    vpc_id = resources[t(cf_igw_attachment["vpc"])]
+                    resource_id = f"{igw_id}:{vpc_id}"
+                elif assoc := item.get("cf_rtassoc"):
+                    resource_id = f"{resources[t(assoc['subnet'])]}/{resources[t(assoc['route_table'])]}"
+                elif bkpsel := item.get("cf_backupselection"):
+                    selection_id, plan_id = resources[t(bkpsel)].split("_")
+                    resource_id = f"{plan_id}|{selection_id}"
+                else:
+                    resource_id = resources[t(item["cf"])]
+                imports.append(f"terraform import '{tf_import_path}' '{resource_id}'")
+
+        print("#!/bin/bash\nset -ex")
+        print("\n".join(imports))
+
+    def create_tfvars(self):
+        self.setup()
+        cdkconfig = yaml.safe_load(
+            [
+                o["OutputValue"]
+                for o in self.cf.describe_stacks(StackName=self.stack_name)["Stacks"][0]["Outputs"]
+                if o["OutputKey"] == "cdkconfig"
+            ][0]
+        )
+
+        def get_subnet_ids(subnet_type: str, prefix: str = "VPC"):
+            return [
+                v
+                for k, v in self.stacks["vpc_stack"]["resources"].items()
+                if re.match(f"{prefix}{self.stack_name}{subnet_type}Subnet\\d+Subnet", k)
+            ]
+
+        ng_role_name = self.stacks["eks_stack"]["resources"][f"{self.stack_name}NG"]
+        client = boto3.client("iam", self.region)
+        ng_role_arn = client.get_role(RoleName=ng_role_name)["Role"]["Arn"]
+        eks_custom_role_maps = [
+            {
+                "rolearn": ng_role_arn,
+                "username": "system:node:{{EC2PrivateDNSName}}",
+                "groups": [
+                    "system:masters",
+                    "system:bootstrappers",
+                    "system:nodes",
+                ],
+            }
+        ]
+
+        tfvars = {
+            "deploy_id": cdkconfig["name"],
+            "region": cdkconfig["aws_region"],
+            "grandfathered_creation_role": self.stacks["eks_stack"]["resources"]["eksCreationRole"],
+            "tags": {**cdkconfig["tags"], "domino-deploy-id": cdkconfig["name"]},
+            "vpc_id": cdkconfig["vpc"]["id"]
+            if not cdkconfig["vpc"]["create"]
+            else self.stacks["vpc_stack"]["resources"]["VPC"],
+            "public_subnet_ids": get_subnet_ids("Public"),
+            "private_subnet_ids": get_subnet_ids("Private"),
+            "pod_subnet_ids": get_subnet_ids("Pod", ""),
+            "k8s_version": cdkconfig["eks"]["version"],  # We're trusting this is accurate
```

Comment:
Is it better to just check & use the EKS cluster data?

Reply:
I'm sure in part I was trying to avoid extra api calls where unnecessary, but I also genuinely am not sure. If there's a mismatch, we could handle that situation in a number of ways (silently do what we think would work, silently do what the config is as above and let terraform be where it's discovered, or do some "mismatch" error with an option to bypass).